kubernetes_asyncio/config/kube_config.py

```diff
@@ -617,16 +619,21 @@ async def refresh_token(loader, client_configuration=None, interval=60):
     :param interval: how often check if token is up-to-date
 
     """
-    if loader.provider != 'gcp':
-        return
-
-    if client_configuration is None:
-        client_configuration = Configuration()
-
-    while 1:
-        await asyncio.sleep(interval)
-        await loader.load_gcp_token()
-        client_configuration.api_key['BearerToken'] = loader.token
+    if loader.provider == 'gcp':
+        if client_configuration is None:
+            client_configuration = Configuration()
+        while 1:
+            await asyncio.sleep(interval)
+            await loader.load_gcp_token()
+            client_configuration.api_key['BearerToken'] = loader.token
+    elif 'exec' in loader._user:
+        if client_configuration is None:
+            client_configuration = Configuration()
+        while 1:
+            await asyncio.sleep(interval)
+            if hasattr(loader, 'expiry') and _is_expired(loader.expiry):
```

Comment:
Yes, I moved the check to inside `load_from_exec_plugin`. 

One thing to note is that I inverted the second clause in the if statement so that the function only has an early return if both the attribute `loader.exec_plugin_expiry` has been set and the token has not expired.